File: benchmarks/poly/fibrec/fibrec.rb
def fib(n)
  return n if n < 2
  fib(n - 1) + fib(n - 2)
end

def main
  _t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
  puts fib(34)
  $stderr.puts "bench_ns=#{Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond) - _t0}"
end

main

File: benchmarks/poly/fibrec/fibrec.js
function fib(n) {
  if (n < 2) return n;
  return fib(n - 1) + fib(n - 2);
}

function main() {
  const _t0 = process.hrtime.bigint();
  console.log(fib(34));
  console.error(`bench_ns=${process.hrtime.bigint() - _t0}`);
}

main();
